src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -1270,6 +1270,29 @@ public static IResourceBuilder<T> WithUrlForEndpoint<T>(this IResourceBuilder<T>
         return builder;
     }
 
+    /// <summary>
+    /// Configures the resource to copy container files from the specified source resource during publishing.
+    /// </summary>
+    /// <typeparam name="T">The type of resource being built. Must implement <see cref="IResource"/>.</typeparam>
+    /// <param name="builder">The resource builder to which container files will be copied to.</param>
+    /// <param name="source">The resource which contains the container files to be copied.</param>
+    /// <param name="destinationPath">The destination path within the resource's container where the files will be copied.</param>
+    public static IResourceBuilder<T> PublishWithContainerFiles<T>(
```

Comment:
If this doesn't just apply in publish mode then I think we call this `WithContainerFiles`.

@danegsta We can make this work in DCP genrally right? (not now but later yes?)